### PR TITLE
Tech/refactor resetting file settings

### PR DESCRIPTION
**Pull request by alschmut**
# Tech/refactor resetting file settings

## Description

- Move resetting/merging the fileSettings (attributeTypes, edges, blacklist, markedPackages `onFileSelectionStateChanged` from storeService to each its own redux service
- This also splits the settingsMerger as each property is now handled inside its `<proeprty>.reset.ts` file inside the `store` directory
- Some common functionality for updating the nodePaths stayed inside the original SettingsMerger
- The SettingsMerger file was renamed to nodePathHelper and contains pure exported functions without a class wrapper

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

- [x] **All** requirements mentioned in the issue are implemented
- [x] Does match the Code of Conduct and the Contribution file
- [x] Task has its own **GitHub issue** (something it is solving)
  - [x] Issue number is **included in the commit messages** for traceability
- [x] **Update the README.md** with any changes/additions made
- [x] **Update the CHANGELOG.md** with any changes/additions made
- [x] **Enough test coverage to ensure that uncovered changes do not break functionality**
- [x] **All tests pass**
- [x] **Descriptive pull request text**, answering:
  - What problem/issue are you fixing?
  - What does this PR implement and how?
- [x] **Assign your PR to someone** for a code review
  - This person _will be contacted **first**_ if a bug is introduced into `master`
- [x] **Manual testing** did not fail

### CODE REVIEW

**Comment by NearW** [comment]
@alschmut requires conflict resolving in order to review

**Comment by alschmut** [comment]
@NearW I merged the current master branch and resolved the conflicts

**Comment by NearW** [comment]
nice refactoring!

I want to disagree with a naming though. I would rename the methods and files from `reset` to `merge` or `merger`, since resetting implies that they go back to some default value. Instead they calculate and merge a new state.

File: visualization/app/codeCharta/state/store/fileSettings/blacklist/blacklist.service.ts [hunk 1]
```diff
@@ -21,6 +25,17 @@ export class BlacklistService implements StoreSubscriber {
 		}
 	}
 
+	public onFileStatesChanged(fileStates: FileState[]) {
+		this.reset(fileStates)
+	}
+
+	public reset(fileStates: FileState[]) {
+		const visibleFiles = FileStateHelper.getVisibleFileStates(fileStates).map(x => x.file)
+		const withUpdatedPath = !!FileStateHelper.isPartialState(fileStates)
```
Comment (NearW) [inline visualization/app/codeCharta/state/store/fileSettings/blacklist/blacklist.service.ts]: Doesn't `isPartialState´ return a boolean? Why would you negate twice here?

Comment (alschmut) [reply visualization/app/codeCharta/state/store/fileSettings/blacklist/blacklist.service.ts]: Yea that actually doesn't make sense :D

**Comment by alschmut** [comment]
> nice refactoring!
> 
> I want to disagree with a naming though. I would rename the methods and files from `reset` to `merge` or `merger`, since resetting implies that they go back to some default value. Instead they calculate and merge a new state.

Cool 👍  I am glad you like it! I wasn't 100% sure about the reset naming as well, and therefor was renaming the methods to `getMerged...()`. I am totally fine with renaming the files and the service methods to `merge` as well

**Comment by NearW** [comment]
In the case of all the arrays being merged here and the previous class being called settingsMERGER, I'd rename the files and methods to merge.

I'll do a manual test once this is done (either by you or by myself) and then merge it.

Thanks for contributing! :D

**Comment by NearW** [comment]
I'm a bit confused.

Is there is reason the attributeTypes file is still `.reset` instead of `.merge`?

And is this functionality even used atm? Whenever the fileSelectionState changes we don't merge at all. Everything is reset. I just excluded a node from sample1 and switched to multiple.
This isn't working on master either, so what is our expected behaviour here?

**Comment by alschmut** [comment]
Oh sorry about the attributeTypes, I just totally forgot.

Regarding the expected (implemented) behaviour:
We do not keep any of the settings, but we merge all fileSettings of the imported files (that are currently is use). So all the fileSettings that are written inside the imported cc.json files

**Pull request merged**